net/api/service.go

```diff
@@ -61,15 +61,20 @@ func (s *Service) GetHostID(_ context.Context, _ *pb.GetHostIDRequest) (*pb.GetH
 func (s *Service) CreateThread(ctx context.Context, req *pb.CreateThreadRequest) (*pb.ThreadInfoReply, error) {
 	log.Debugf("received create thread request")
 
-	creds, err := getCredentials(req.Credentials)
+	id, err := thread.Cast(req.ThreadID)
 	if err != nil {
-		return nil, err
+		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 	opts, err := getKeyOptions(req.Keys)
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
-	info, err := s.net.CreateThread(ctx, creds, opts...)
+	auth, err := thread.NewAuthFromMD(ctx)
```

Comment:
Here we can pull out the auth from the metadata and simply pass to the internal API.